core/src/main/java/hudson/model/Actionable.java

```diff
@@ -116,7 +130,11 @@
      * @see #getAction(Class)
      */
     public <T extends Action> List<T> getActions(Class<T> type) {
-        return Util.filter(getAllActions(), type);
+        List<T> _actions = Util.filter(getActions(), type);
+        for (TransientActionFactory<?> taf : TransientActionFactory.factoriesFor(getClass(), type)) {
+            _actions.addAll(Util.filter(createFor(taf), type));
+        }
+        return Collections.unmodifiableList(_actions);
```

Comment:
Is this improvement really worth dealing with potential breakage in whatever code is dumb enough to modify the returned list?